manifests/server/wait_online.pp

```diff
@@ -0,0 +1,16 @@
+# @summary Creates a dummy exec to allow deferring applies until the Kubernetes API server has started
+class k8s::server::wait_online {
+  # Wait up to 30 seconds for kube-apiserver to start
+  exec { 'k8s-apiserver wait online':
+    command     => 'kubectl --kubeconfig /root/.kube/config version',
+    path        => $facts['path'],
+    refreshonly => true,
+    tries       => 15,
+    try_sleep   => 2,
+  }
+
+  # Require possibly managed components before checking online state
+  Kubeconfig <| title == '/root/.kube/config' |> -> Exec['k8s-apiserver wait online']
+  K8s::Binary <| title == 'kubectl' |> -> Exec['k8s-apiserver wait online']
+  Service <| title == 'kube-apiserver' |> ~> Exec['k8s-apiserver wait online']
```

Comment:
why differentiate between kube-apiserver and k8s-apiserver? I just want to understand the naming 😄

Reply:
Well, kube-apiserver is the name of the apiserver application, k8s-apiserver is for the apiserver component as managed by the module - there's technically no fixed link between the two as the apiserver can run as an in-cluster component when self-hosted. The dash there is probably a bit superfluous though.